### PR TITLE
Datepicker: add Calendar Icon

### PR DESCRIPTION
**Pull request by Zombiefruit**
Added a calendar icon that can be clicked to toggle the calendar

### CODE REVIEW

File: src/components/date-picker/date-picker-icons.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+
+export const CalendarIcon: React.SFC<React.HTMLAttributes<SVGElement>> = props => {
+    return (
+        <svg
```
Comment (JoMarton) [inline src/components/date-picker/date-picker-icons.tsx]: if this SVG element should not get focus add attribute `focusable="false"` for IE

Comment (JoMarton) [reply src/components/date-picker/date-picker-icons.tsx]: most of these attributes can and should be moved to the css class

File: src/components/date-picker/date-picker.tsx [hunk 2]
```diff
@@ -17,6 +18,7 @@ export interface DatePickerProps extends FormInputProps<Date> {
     readOnly?: boolean;
     showDropdownOnInit?: boolean;
     startingDay?: number;
+    calendarIcon?: React.SFC;
```
Comment (JoMarton) [inline src/components/date-picker/date-picker.tsx]: please use `React.ComponentType`

File: src/components/date-picker/date-picker.tsx [hunk 3]
```diff
@@ -66,6 +68,14 @@ export class DatePicker extends React.PureComponent<DatePickerProps, DatePickerS
                     type="text"
                     data-automation-id="DATE_PICKER_INPUT"
                 />
+                {this.props.calendarIcon
+                    ? this.props.calendarIcon
+                    : <CalendarIcon
+                        className="icon"
+                        data-automation-id="CALENDAR_ICON"
+                        onMouseDown={this.toggleDropdown}
```
Comment (AviVahl) [inline src/components/date-picker/date-picker.tsx]: why not click?

File: src/components/date-picker/date-picker.tsx [hunk 4]
```diff
@@ -66,6 +68,14 @@ export class DatePicker extends React.PureComponent<DatePickerProps, DatePickerS
                     type="text"
                     data-automation-id="DATE_PICKER_INPUT"
                 />
+                {this.props.calendarIcon
```
Comment (AviVahl) [inline src/components/date-picker/date-picker.tsx]: default props
rename prop name when destructuring

**Pull request merged**